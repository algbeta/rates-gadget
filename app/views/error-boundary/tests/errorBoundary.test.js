import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ErrorBoundary from '../index';

describe('ErrorBoundary tests', () => {
  test('ErrorBoundary renders correctly', () => {
    const tree = renderer
      .create(
        <ErrorBoundary>
          <div>Child</div>
        </ErrorBoundary>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ErrorBoundary with error', () => {
    const NestComponent = () => null;

    const wrapper = shallow(
      <ErrorBoundary>
        <NestComponent />
      </ErrorBoundary>,
    );

    wrapper.find('NestComponent').simulateError(new Error('error'));
    expect(wrapper.state()).toEqual({ hasError: true });
  });
});
