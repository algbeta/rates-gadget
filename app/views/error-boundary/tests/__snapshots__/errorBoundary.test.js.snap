// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary tests ErrorBoundary renders correctly 1`] = `
<div>
  Child
</div>
`;
